FWCore/ServiceRegistry/interface/ServiceRegistry.h

```diff
@@ -61,9 +61,10 @@ namespace edm {
     template <typename T>
     T& get() const {
       if (nullptr == manager_.get()) {
-        Exception::throwThis(errors::NotFound,
-                             "Service"
-                             " no ServiceRegistry has been set for this thread");
+        Exception::throwThis(
+            errors::NotFound,
+            "No ServiceRegistry has been set for this thread. Tried to get a Service with compiler type name ",
+            typeid(T).name());
```

Comment:
depending what they look like, would it make sense to demangle these ?
```suggestion
            boost::core::demangle(typeid(T).name()));
```

Reply:
Thanks for the suggestion. While this particular exception points to a framework bug, there is another one in `ServicesManager.h` that is thrown if a requested Service is missing from the configuration. What I recall from actually seeing that exception, the mangled name has usually(?) been informative enough to point to right direction, but a demangled name would certainly be more clear. I changed the two exceptions here as well as the one in `ServicesManager` to use demangled name.